tests/server/test_base_server.py

```diff
@@ -14,11 +17,19 @@
 from fl4health.client_managers.poisson_sampling_manager import PoissonSamplingClientManager
 from fl4health.parameter_exchange.full_exchanger import FullParameterExchanger
 from fl4health.server.base_server import FlServer, FlServerWithCheckpointing
+from fl4health.strategies.basic_fedavg import BasicFedAvg
+from fl4health.utils.metrics import TestMetricPrefix
+from tests.test_utils.custom_client_proxy import CustomClientProxy
 from tests.test_utils.models_for_test import LinearTransform
 
 model = LinearTransform()
 
 
+def metrics_aggregation(to_aggregate: List[Tuple[int, Metrics]]) -> Metrics:
```

Comment:
So I know it's a bit of a pain, but let's try using the `evaluate_metrics_aggregation_fn` in `fl4health/utils/metric_aggregation.py` and passing a list with two different EvaluateRes objects. That way we actually see `_handle_result_aggregation` actually extract and computed a weighted average correctly based on num_examples for both a validation and test set.

Reply:
resolved in the latest commit